### PR TITLE
feat: add max_cols option in segment

### PR DESCRIPTION
**Pull request by mirsella**
If the terminal width exceeds this value, the segment will be hidden.
Need testing on Windows.

Follow up of discussion https://github.com/JanDeDobbeleer/oh-my-posh/discussions/3148

I hope the doc changes are good.
thanks !

### CODE REVIEW

**Comment by mirsella** [comment]
image

i still need to figure out why the RPROMPT is on the newline, i want it on the first line with the PROMPT. if you have a guess i'm taking. thanks

**Comment by JanDeDobbeleer** [comment]
@mirsella rprompt will ALWAYS be on the same line as the cursor. That's by design.

**Comment by mirsella** [comment]
oh ok, thanks for clearing things up

File: src/engine/segment.go [hunk 1]
```diff
@@ -399,6 +402,9 @@ func (segment *Segment) SetEnabled(env platform.Environment) {
 			}
 		}
 	}
+	if width, _, err := term.GetSize(int(os.Stdin.Fd())); err == nil && segment.MaxCols > 0 && width > segment.MaxCols {
```
Comment (JanDeDobbeleer) [inline src/engine/segment.go]:
```suggestion
	if width, err := segment.env.TerminalWidth(); err == nil && segment.MaxCols > 0 && width > segment.MaxCols {
```
We actually have this number already for other means so no use for a specific library here. Make sure to use `go mod tidy` after applying this change to remove the dependencies.

File: website/docs/configuration/segment.mdx [hunk 2]
```diff
@@ -50,7 +50,8 @@ understand how to configure a segment.
...
-| `alias`                | `string`     | for use with [cross segment template properties][cstp]                                                                                                                                                                                                                                                                            |
+| `alias`                | `string`     | for use with [cross segment template properties][cstp]                                                                                                                                                                                                                                                                     |
+| `max_cols`             | `int`        | if the terminal width exceeds this value, the segment will be hidden. for your terminal width, see `tput cols`. Defaults to `0`
```
Comment (JanDeDobbeleer) [inline website/docs/configuration/segment.mdx]: `tput cols` is a unix functionality, I would replace that by adding a `oh-my-posh get cols` [command](https://github.com/JanDeDobbeleer/oh-my-posh/blob/d34d9d368c1b74b12734dfb552ab711ab7d0ade1/src/cli/get.go#L46).

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

**Comment by JanDeDobbeleer** [comment]
General question, do we want to add a `min_cols` setting as well? if so it can all become very responsive.

**Comment by mirsella** [comment]
I'll look into it tomorrow when I get back to my pc.
we can add a min_cols while we are at it

**Comment by JanDeDobbeleer** [comment]
@mirsella we've got time. I'm not in a hurry.

**Comment by mirsella** [comment]
Tell me if there is anything else !

**Comment by mirsella** [comment]
Accidentally merged your main branch with this PR but the commit message wasn't conventional, so i deleted it

**Review by JanDeDobbeleer — APPROVED** [review]

**Comment by JanDeDobbeleer** [comment]
@all-contributors please add @mirsella for code,design,doc

**Pull request merged**